core/tests/protractor_utils/ExplorationPlayerPage.js

```diff
@@ -80,6 +80,49 @@ var ExplorationPlayerPage = function() {
   var suggestionPopupLink =
     element(by.css('.protractor-test-exploration-suggestion-popup-link'));
 
+  var audioBar = element(
+    by.css('.protractor-test-audio-bar'));
+  var audioSelect = element(
+    by.css('.audio-language-select'));
+
+  this.clickAudioBar = function() {
+    waitFor.elementToBeClickable(audioBar,
+      'Audio bar taking too long to be clickable');
+    audioBar.click();
+  };
+
+  this.playAudio = function() {
+    var playBtn = element(
+      by.css('.fa-play-circle'));
+    waitFor.elementToBeClickable(playBtn,
+      'Play button taking too long to be clickable');
+    playBtn.click();
+  };
+
+  this.expectAudioToBePlaying = function() {
+    var pauseBtn = element(
+      by.css('.fa-pause-circle'));
+    expect(pauseBtn.isPresent()).toBeTruthy();
+  };
+
+  this.pauseAudio = function() {
+    var pauseBtn = element(
+      by.css('.fa-pause-circle'));
+    waitFor.elementToBeClickable(pauseBtn,
+      'Pause button taking too long to be clickable');
+    pauseBtn.click();
+  };
+
+  this.expectAudioToBePaused = function() {
+    var playBtn = element(
+      by.css('.fa-play-circle'));
+    expect(playBtn.isPresent()).toBeTruthy();
```

Comment:
Could we add some check for the actual audio output?

Reply:
Hmm, I thought that the icon changing asserted that the audio was playing?

In `audio-bar.directive.html` there is:

`ng-class="{'fa-ellipsis-h': audioLoadingIndicatorIsShown, 'fa-play-circle protractor-test-play-circle': !isAudioPlaying(), 'fa-pause-circle protractor-test-pause-circle': isAudioPlaying(), 'audio-controls-audio-not-available': !isAudioAvailableInCurrentLanguage() || audioIsLoading,}`

So if the icon changes doesn't it also assert that audio is playing/not playing?

Reply:
It is probably closely related but it would still be nice to check the actual audio output (if possible) in case of some error in the audio service that we use.